### DISCUSSION TITLE
상태관리관련하여 server state와  client state의 차이점을 설명해보세요.

### DISCUSSION THREAD

**Discussion by bona373737** [wanted-9th-3team/frontend-study#20] [프론트 기술 질문 공간]
server state와  client state의 차이점을 설명해보세요.

**Comment by michoball** [comment]
server state는 클라이언트에 표시하기위해 서버에서 받아온 데이터를 말합니다.  특정 시점의 서버 데이터를 가져와 사용하기 때문에 비동기적인 상태를 가지고 있습니다. 

client state는 클라이언트 안에서 생성되고 사용되는 데이터를 말합니다. 어플에 전역으로 사용되는 state와 컴포넌트 안에서 사용되는 state가 있습니다. 

client state나 server state 데이터를 앱에서 전역으로 다루기 위해 redux와 같은 상태관리 라이브러리를 많이 사용했는데 최근에는 server state의 데이터를 다루는 툴로 react-qeury , swr과 같은 툴을 사용하고 있습니다.

**Comment by WooGie911** [comment]
Client State란 클라이언트가 자체적으로 생성한 상태로 사용자의 input 값이나 UI 변경을 담당하는 부분으로 볼 수 있습니다. 
쉽게 말해 사용자가 사용하는 FE 영역이라고 볼 수 있습니다.
즉, 서버에서 일어나는 일과는 아무 관련이 없는 상태들을 의미합니다.

반면 Server State란 클라이언트에 표시하는 데 필요한 서버 데이터를 의미합니다. 
쉽게 말해 DB에 존재하는 데이터들. 즉 BE 영역에 가깝다고 볼 수 있습니다.

**Comment by skyhanull** [comment]
Client State 는 모달에 대한 isOpen 와 같은 UI State, 폼 관련된 Form State 그리고 URL State 등을 의미한다.
Server State 는 API 의 응답값을 의미한다.
이부분에 대해선 애매합니다..

**Comment by BaekYuri** [comment]
Server State는 서버에서 요청해서 받아오는 데이터를 의미합니다. 클라이언트에 저장한 Server State와 관계 없이, 서버의 데이터는 수시로 변하기 때문에 적절히 다루기 위해서는 오래된 데이터를 업데이트하는 기능이 필요합니다.
Client State는 클라이언트 자체에서 갖는 상태입니다. 예를 들어 사용자의 input 값이나 UI 변경 관련 값 등이 있습니다.

**Comment by bona373737** [comment]
Client State란 클라이언트가 자체적으로 생성한 상태로 사용자의 input 값이나 UI 변경을 담당하는 부분으로 볼 수 있습니다. 예를 들어, 모달창을 열고 닫는 상태나 사용자가 입력한 값 같은 것들을 의미합니다.

Server State는 클라이언트에 표시하는 데 필요한 서버 데이터를 의미합니다.데이터를 가져오고 업데이트하기 위해선 비동기 API가 필요합니다.

최근엔 서버로부터 가져온 데이터를 react-query를 사용해 관리하고 데이터를 가공해서 클라이언트측에서 전역적으로 보관하고 있을 데이터를 redux, recoil등의 전역상태관리라이브러리를 사용하여 분리하여 관리하는 모습들을 찾아 볼 수 있었습니다.
